document-store/src/integrationTest/java/org/hypertrace/core/documentstore/DocStoreTest.java

```diff
@@ -1487,6 +1492,34 @@ public void testSearchIteratorInterface(String dataStoreName) throws IOException
     }
   }
 
+  @ParameterizedTest
+  @MethodSource("databaseContextProvider")
+  public void testNewAggregateApiWhereClause(String dataStoreName) throws IOException {
+    Map<Key, Document> documents = createDocumentsFromResource("mongo/collection_data.json");
+    Datastore datastore = datastoreMap.get(POSTGRES_STORE);
+    Collection collection = datastore.getCollection(COLLECTION_NAME);
+
+    // add docs
+    boolean result = collection.bulkUpsert(documents);
+    Assertions.assertTrue(result);
+
+    // query docs
+    org.hypertrace.core.documentstore.query.Query query =
+        org.hypertrace.core.documentstore.query.Query.builder()
+            .setFilter(
+                RelationalExpression.of(
+                    IdentifierExpression.of("quantity"), NEQ, ConstantExpression.of(10)))
+            .build();
+
+    Iterator<Document> resultDocs = collection.aggregate(query);
+    int count = 0;
+    while (resultDocs.hasNext()) {
+      count++;
+      resultDocs.next();
+    }
+    Assertions.assertEquals(6, count);
```

Comment:
Maybe we could have stricter checks (like the exact documents instead of just the count) in a different PR.